### PR TITLE
Screens for complaint feedback and room booking

### PR DESCRIPTION
**Pull request by Hardik-2311**

### CODE REVIEW

**Comment by Hardik-2311** [comment]
@beingnoble03 kindly review

**Comment by Hardik-2311** [comment]
https://github.com/IMGIITRoorkee/omniport-frontend-bhawan-app/assets/126074137/f6b4d96a-2118-40a6-917e-f57f6608e27a

File: src/components/feebback-card-new/index.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+                            <td className={tailwindWrapper("border px-4 py-2 col-span-2")}>Some feedback text</td>
+                            <td className={tailwindWrapper("border px-4 py-2")}>Attachment</td>
+                            <td className={tailwindWrapper("border px-4 py-2")}>Pending</td>
+                            <td className={tailwindWrapper("border px-4 py-2")}>2024-03-29</td>
+                        </tr>
+                        <tr className={tailwindWrapper("grid grid-cols-5")}>
+                            <td className={tailwindWrapper("border px-4 py-2 col-span-2")}>Some feedback text</td>
+                            <td className={tailwindWrapper("border px-4 py-2")}>Attachment</td>
+                            <td className={tailwindWrapper("border px-4 py-2")}>Pending</td>
+                            <td className={tailwindWrapper("border px-4 py-2")}>2024-03-29</td>
```
Comment (beingnoble03) [inline src/components/feebback-card-new/index.js:34]: dummy data?

File: src/components/events-new/index.js [hunk 2]
```diff
@@ -8,44 +10,92 @@ class NewEvents extends Component {
...
+
+  componentDidUpdate(prevProps) {
+    if (prevProps.activeHostel !== this.props.activeHostel) {
+      this.fetchEvents();
+    }
   }
 
+  fetchEvents = async () => {
+    try {
+      console.log(this.props.activeHostel)
```
Comment (beingnoble03) [inline src/components/events-new/index.js:32]: remove this statement

File: src/components/complaints-new/index.js [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+    };
+
+    render() {
+        return (
+            <div className={tailwindWrapper("w-2/3 ml-16")}>
+
+                <form onSubmit={this.handleSubmit} className={tailwindWrapper("  mb-8")}>
+                    <div className={tailwindWrapper("mb-4")}>
+                        <Link to="/bhawan_app/complaints" className={tailwindWrapper("text-[#133BC5] font-bold text-xl")}>Complaints</Link>
+                        <span>{"   |   "}</span>
```
Comment (beingnoble03) [inline src/components/complaints-new/index.js:82]: why is this `"   |   "`? i think we should use padding or margin instead.

File: src/components/book-room-new/index.js [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+                            )}
+                        </div>
+                    ))}
+                    <div className={tailwindWrapper("my-8")}>
+                        <button type="button" onClick={this.increaseVisitor} className={tailwindWrapper('text-[#0B2274] text-xl font-bold')}>
+                            + Add Visitor
+                        </button>
+                    </div>
+
+                    {/* Submit Button */}
```
Comment (beingnoble03) [inline src/components/book-room-new/index.js:276]: remove this comment

File: src/components/book-room-new/index.js [hunk 5]
```diff
@@ -0,0 +1,304 @@
...
+    };
+
+    render() {
+        const { loading, fromDate, endDate, visitors } = this.state;
+
+        return (
+            <div className={tailwindWrapper('w-2/3')}>
+                {/* Form */}
+                <form onSubmit={this.handleSubmit} className={tailwindWrapper('px-8 pt-6 pb-8 mb-4')}>
+                    {/* Date Inputs */}
```
Comment (beingnoble03) [inline src/components/book-room-new/index.js:126]: remove this

File: src/components/book-room-new/index.js [hunk 6]
```diff
@@ -0,0 +1,304 @@
...
+            this.errCallBack
+        );
+    };
+
+    render() {
+        const { loading, fromDate, endDate, visitors } = this.state;
+
+        return (
+            <div className={tailwindWrapper('w-2/3')}>
+                {/* Form */}
```
Comment (beingnoble03) [inline src/components/book-room-new/index.js:124]: remove

File: src/components/book-room-new/index.js [hunk 7]
```diff
@@ -0,0 +1,304 @@
...
+            moment(this.state.fromDate, 'YYYY-MM-DD').format('YYYY-MM-DD')
+        );
+        formData.append(
+            'requestedTill',
+            moment(this.state.endDate, 'YYYY-MM-DD').format('YYYY-MM-DD')
+        );
+
+        // Iterate over visitors array and append visitor data to formData
+        this.state.visitors.forEach((visitor, index) => {
+            console.log(index)
```
Comment (beingnoble03) [inline src/components/book-room-new/index.js:77]: remove

File: src/components/book-room-new/index.js [hunk 8]
```diff
@@ -0,0 +1,304 @@
...
+        this.setState({ visitors });
+    };
+
+    handleFileInputChange = (event, index) => {
+        const file = event.target.files[0];
+        this.setState({ selectedFile: file });
+    };
+
+    handleSubmit = (e) => {
+        console.log(this.state.fromDate, this.state.endDate, this.state.visitors);
```
Comment (beingnoble03) [inline src/components/book-room-new/index.js:58]: remove

**Review by beingnoble03 — CHANGES_REQUESTED** [review]

**Comment by Hardik-2311** [comment]
@beingnoble03  done changes

File: src/components/Feedback/index.js [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+
+        this.setState({
+            [name]: value
+        });
+    }
+
+    handleSubmit = (event) => {
+        event.preventDefault();
+        alert("Feedback submitted");
+        console.log(this.state);
```
Comment (beingnoble03) [inline src/components/Feedback/index.js:36]: remove

File: src/components/book-room-new/index.js [hunk 10]
```diff
@@ -0,0 +1,298 @@
...
+            if (this.state.proof && this.state.proof[index]) {
+                formData.append(
+                    `visitors[${index}][proof]`,
+                    this.state.proof[index],
+                    this.state.proof[index].name
+                );
+            }
+
+        });
+
```
Comment (beingnoble03) [inline src/components/book-room-new/index.js:98]: extra line? lint with prettier once

File: src/components/book-room-new/index.js [hunk 11]
```diff
@@ -0,0 +1,298 @@
...
+                                type='date'
+                                id='endDate'
+                                name='endDate'
+                                value={endDate}
+                                onChange={this.handleChange}
+                                className={tailwindWrapper('border rounded  py-4 px-10 text-gray-700 leading-tight focus:outline-none focus:shadow-outline')}
+                            />
+                        </div>
+                    </div>
+
```
Comment (beingnoble03) [inline src/components/book-room-new/index.js:149]: extra line?

File: src/components/book-room-new/index.js [hunk 12]
```diff
@@ -0,0 +1,298 @@
...
+                                            name="relation"
+                                            value={visitor.relation}
+                                            onChange={(e) => this.handleChange(e, index)}
+                                            placeholder="Relation"
+                                            className={tailwindWrapper('appearance-none border rounded py-4 px-6 text-gray-700 leading-tight focus:outline-none focus:shadow-outline')}
+                                        />
+                                    </div>
+                                    <div className={tailwindWrapper("flex flex-col")}>
+                                        <label htmlFor='proof' className={tailwindWrapper('block text-[#0B2274] text-sm font-bold mb-2')}>
+                                            Identiti Proof
```
Comment (beingnoble03) [inline src/components/book-room-new/index.js:211]: typo

File: src/components/feebback-card-new/index.js [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+                    <thead className={tailwindWrapper("p-12 bg-[#6381EB] text-white")}>
+                        <tr className={tailwindWrapper("grid grid-cols-5 px-6 py-4")}>
+                            <td className={tailwindWrapper("px-4 py-2 col-span-2 ")}>Feedbacks</td>
+                            <td className={tailwindWrapper("px-4 py-2 ")}>Attachments</td>
+                            <td className={tailwindWrapper("px-4 py-2 ")}>Status</td>
+                            <td className={tailwindWrapper("px-4 py-2 ")}>Date & time</td>
+                        </tr>
+                    </thead>
+                    <tbody >
+                        {/* real data will be done after feedback actions */}
```
Comment (beingnoble03) [inline src/components/feebback-card-new/index.js:18]:
```suggestion
                        {/* TODO: real data will be done after feedback actions */}
```

**Review by beingnoble03 — CHANGES_REQUESTED** [review]

**Review by beingnoble03 — APPROVED** [review]

**Pull request merged**